### PR TITLE
Additional Board: Robo HAT MM1 board

### PR DESCRIPTION
**Pull request by wallarug**
Ready to push first iteration of Robo HAT MM1 board.

The Robo HAT MM1 is developed and maintained by the Robotics Masters team.   Information about the board can be found on Crowd Supply.

https://www.crowdsupply.com/robotics-masters/robo-hat-mm1

Must be Merged at same time as [PR 1075](https://github.com/microsoft/pxt-common-packages/pull/1075) in the `microsoft/pxt-common-packages` repository

### CODE REVIEW

**Comment by wallarug** [comment]
@peterpanstechland  can also verify working.

File: docs/boards.md [hunk 1]
```diff
@@ -37,9 +37,9 @@ This editor supports various maker boards, based on SAMD21, SAMD51, NRF52.
         "imageUrl": "/static/libs/redbear-ble-nano.jpg"  
     },
     {
-        "name": "Add your board",
-        "url": "/boards/add-a-new-board",
-        "imageUrl": "/static/libs/bead-servo.jpg"
+        "name": "Robotics Masters Robo HAT MM1 M4",
```
Comment (pelikhan) [inline docs/boards.md]: Leave this card and add a new one.

File: libs/robotics-masters-robohatmm1-m4/device.d.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    //% fixedInstance shim=pxt::getPin(PIN_SERVO5)
+    const SERVO5: PwmPin;
+    //% fixedInstance shim=pxt::getPin(PIN_SERVO6)
+    const SERVO6: PwmPin;
+    //% fixedInstance shim=pxt::getPin(PIN_SERVO7)
+    const SERVO7: PwmPin;
+    //% fixedInstance shim=pxt::getPin(PIN_SERVO8)
+    const SERVO8: PwmPin;
+
+    //% fixedInstance shim=pxt::getPin(PIN_SERVO7)
```
Comment (pelikhan) [inline libs/robotics-masters-robohatmm1-m4/device.d.ts]: copy paste issue

File: libs/robotics-masters-robohatmm1-m4/device.d.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    const GROVE0: PwmPin;
+    //% fixedInstance shim=pxt::getPin(PIN_SERVO8)
```
Comment (pelikhan) [inline libs/robotics-masters-robohatmm1-m4/device.d.ts]: copy paste

File: libs/robotics-masters-robohatmm1-m4/device.d.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    const PI_RX: DigitalInOutPin;
+    //% fixedInstance shim=pxt::getPin(PIN_TX)
+    const PI_TX: DigitalInOutPin;
+
+    //% fixedInstance shim=pxt::getPin(PIN_SDA)
+    const SDA: DigitalInOutPin;
+    //% fixedInstance shim=pxt::getPin(PIN_SCL)
+    const SCL: DigitalInOutPin;
+
+    //% fixedInstance shim=pxt::getPin(PIN_SDA)
```
Comment (pelikhan) [inline libs/robotics-masters-robohatmm1-m4/device.d.ts]: copy paste

File: libs/robotics-masters-robohatmm1-m4/device.d.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+}
+
+declare namespace input {
+    /**
+     * Left button.
+     */
+    //% indexedInstanceNS=input indexedInstanceShim=pxt::getButton
+    //% block="button A" weight=95 fixedInstance
+    //% shim=pxt::getButton(0)
```
Comment (pelikhan) [inline libs/robotics-masters-robohatmm1-m4/device.d.ts]: 0 ?

File: libs/robotics-masters-robohatmm1-m4/ns.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
+
```
Comment (pelikhan) [inline libs/robotics-masters-robohatmm1-m4/ns.ts]: we've deprecated having custom ns.ts files, please remove.

File: libs/robotics-masters-robohatmm1-m4/pxt.json [hunk 7]
```diff
@@ -0,0 +1,55 @@
+{
+    "name": "robotics-masters-robohatmm1-m4",
+    "description": "Support for Robotics Masters Robo HAT MM1 M4",
+    "files": [
+        "README.md",
+        "device.d.ts",
+        "config.ts",
+        "ns.ts",
```
Comment (pelikhan) [inline libs/robotics-masters-robohatmm1-m4/pxt.json]: remove this file

File: package-lock.json [hunk 8]
```diff
@@ -0,0 +1,4962 @@
+{
```
Comment (pelikhan) [inline package-lock.json]: delete this file

**Comment by wallarug** [comment]
Will do.  Thanks.

File: libs/robotics-masters-robohatmm1-m4/device.d.ts (hunk 2 above)
Comment (wallarug) [reply libs/robotics-masters-robohatmm1-m4/device.d.ts]: resolved by b6077cc

File: libs/robotics-masters-robohatmm1-m4/device.d.ts (hunk 3 above)
Comment (wallarug) [reply libs/robotics-masters-robohatmm1-m4/device.d.ts]: Resolved by b6077cc

File: libs/robotics-masters-robohatmm1-m4/device.d.ts (hunk 4 above)
Comment (wallarug) [reply libs/robotics-masters-robohatmm1-m4/device.d.ts]: Resolved by b6077cc

File: libs/robotics-masters-robohatmm1-m4/ns.ts (hunk 6 above)
Comment (wallarug) [reply libs/robotics-masters-robohatmm1-m4/ns.ts]: Sure.  No problem.  

Resolved by b6077cc

File: libs/robotics-masters-robohatmm1-m4/pxt.json (hunk 7 above)
Comment (wallarug) [reply libs/robotics-masters-robohatmm1-m4/pxt.json]: Resolved by b6077cc

File: package-lock.json (hunk 8 above)
Comment (wallarug) [reply package-lock.json]: Resolved by 48e47c6

File: docs/boards.md (hunk 1 above)
Comment (wallarug) [reply docs/boards.md]: Not 100% sure this is fixed correctly.  Feedback required.

File: libs/robotics-masters-robohatmm1-m4/device.d.ts (hunk 5 above)
Comment (wallarug) [reply libs/robotics-masters-robohatmm1-m4/device.d.ts]: Resolved by b6077cc

**Comment by wallarug** [comment]
Hey @pelikhan  there is only that one item to check.   

We are working through a few issues at the moment.   If you could please leave this open as is and I can ping you again when ready.

**Current Issues**

- [Serial USB](https://makecode.microbit.org/device/serial) (not sure how to do this atm)
- Serial over PI_TX/PI_RX is not working but works with Metro M4 D11/D12 ?
- IMU needs additional library in codal-core (future improvement)

File: docs/projects/SUMMARY.md [hunk 9]
```diff
@@ -58,10 +58,3 @@
   * [controller button](/projects/jacdac/controller-button)
   * [LCD screen](/projects/jacdac/lcd-screen)
 * [Boards](/boards)
-  * [Adafruit](/boards/adafruit)
```
Comment (pelikhan) [inline docs/projects/SUMMARY.md]: undo changes

File: docs/projects.md [hunk 10]
```diff
@@ -95,8 +95,7 @@
     },
     {
         "name": "Boards",
-        "url": "/boards",
-        "imageUrl": "/static/libs/adafruit-circuit-playground-express.jpg"
+        "url": "/boards"
```
Comment (pelikhan) [inline docs/projects.md]: undo changes

File: libs/robotics-masters-robohatmm1-m4/device.d.ts [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+}
+
+declare namespace input {
+    /**
+     * Left button.
+     */
+    //% indexedInstanceNS=input indexedInstanceShim=pxt::getButton
+    //% block="button A" weight=95 fixedInstance
+    //% shim=pxt::getButton(BTN_A)
```
Comment (pelikhan) [inline libs/robotics-masters-robohatmm1-m4/device.d.ts]: what is BTN_A?

File: libs/robotics-masters-robohatmm1-m4/pxt.json [hunk 12]
```diff
@@ -0,0 +1,54 @@
+{
+    "name": "robotics-masters-robohatmm1-m4",
+    "description": "Support for Robotics Masters Robo HAT MM1 M4",
```
Comment (pelikhan) [inline libs/robotics-masters-robohatmm1-m4/pxt.json]: add "- beta" in description until all issues are resolved.

File: libs/robotics-masters-robohatmm1-m4/device.d.ts (hunk 11 above)
Comment (wallarug) [reply libs/robotics-masters-robohatmm1-m4/device.d.ts]: There is a button on the Robo HAT MM1.  It is on the left side connected to PA27

Comment (pelikhan) [reply libs/robotics-masters-robohatmm1-m4/device.d.ts]: Is this pin defined in device.d.ts?

Comment (wallarug) [reply libs/robotics-masters-robohatmm1-m4/device.d.ts]: It was incorrectly mapped to a different pin.  I'm not to sure about device.d.ts.

The way the button is defined seems very different to other boards.

Metro M4
```
/**
     * Button connecting D12 to GND.
     */
    //% block="button D12" fixedInstance
    //% shim=pxt::getButtonByPin(PIN_D12,BUTTON_ACTIVE_LOW_PULL_UP)
    //% parts="buttons"
```

Robo HAT Definition
```
    /**
     * Left button.
     */
    //% indexedInstanceNS=input indexedInstanceShim=pxt::getButton
    //% block="button A" weight=95 fixedInstance
    //% shim=pxt::getButton(BTN_A)
    const buttonA: Button;
```

Should I just update to the same format as Metro M4?

**Comment by wallarug** [comment]
Hi @pelikhan,

I hope you have had a good week.   We have sorted out all the bugs and things.  We left the beta tag in.

We are ready for another review and then to merge.

Thank you

File: libs/robotics-masters-robohatmm1-m4/device.d.ts (hunk 11 above)
Comment (wallarug) [reply libs/robotics-masters-robohatmm1-m4/device.d.ts]: Resolved in latest commit.

**Comment by wallarug** [comment]
Hey @pelikhan ,

Just checking in on this one.   There will not be any more changes to this branch now, pending the PR review.

Thanks for all your hard work! 👍

File: libs/core---samd51/dal.d.ts [hunk 13]
```diff
@@ -1051,6 +1051,27 @@ declare const enum DAL {
     CFG_PIN_LCD_DATALINE7 = 1014,
     CFG_NUM_LCD_COLUMNS = 1015,
     CFG_NUM_LCD_ROWS = 1016,
+    CFG_PIN_RCC1 = 1017,
+    CFG_PIN_RCC2 = 1018,
+    CFG_PIN_RCC3 = 1019,
+    CFG_PIN_RCC4 = 1020,
+    CFG_PIN_SERVO1 = 1021,
```
Comment (pelikhan) [inline libs/core---samd51/dal.d.ts]: our pin sequences typically start at 0, please add CFG_PIN_SERVO0

**Pull request merged**